test/fuzz/signature/signature.go

```diff
@@ -57,3 +60,88 @@ func FuzzECDSASigner(data []byte) int {
 
 	return 1
 }
+
+func FuzzComputeDigest(data []byte) int {
+	ecdsaSupportedHashFuncs := []crypto.Hash{
+		crypto.SHA256,
+		crypto.SHA512,
+		crypto.SHA384,
+		crypto.SHA224,
+		crypto.SHA1,
+	}
+	data, _, err := signature.ComputeDigestForSigning(bytes.NewReader(data), crypto.SHA512, ecdsaSupportedHashFuncs)
+	if err != nil {
+		if data != nil {
+			panic(fmt.Sprintf("key %v is not nil when there is an error %v ", data, err))
+		}
+		return 0
+	}
+	return 1
+}
+
+func FuzzComputeVerifying(data []byte) int {
+	ecdsaSupportedHashFuncs := []crypto.Hash{
+		crypto.SHA256,
+		crypto.SHA512,
+		crypto.SHA384,
+		crypto.SHA224,
+		crypto.SHA1,
+	}
+	data, _, err := signature.ComputeDigestForVerifying(bytes.NewReader(data), crypto.SHA512, ecdsaSupportedHashFuncs)
+	if err != nil {
+		if data != nil {
+			panic(fmt.Sprintf("key %v is not nil when there is an error %v ", data, err))
+		}
+		return 0
+	}
+	return 1
+}
+
+func FuzzED25529SignerVerfier(data []byte) int {
+	x := ed25519.PrivateKey(data)
+
+	signer, err := signature.LoadED25519SignerVerifier(x)
+	if err != nil {
+		if signer != nil {
+			panic(fmt.Sprintf("key %v is not nil when there is an error %v ", signer, err))
+		}
+		return 0
+	}
+
+	sig, err := signer.SignMessage(bytes.NewReader(data))
+	if err != nil {
+		if sig != nil {
+			panic(fmt.Sprintf("key %v is not nil when there is an error %v ", sig, err))
+		}
+		return 0
+	}
+
+	signer.VerifySignature(bytes.NewReader(sig), bytes.NewReader(data))
+	return 1
+}
+
+func FuzzRSASignerVerfier(data []byte) int {
```

Comment:
`s/RSA/RSAPKCS1v15` to make the function name more explicit?

Reply:
Renamed the function.